db/database.go

```diff
@@ -152,8 +155,12 @@ func ConnectToBucket(spec base.BucketSpec, callback func(bucket string, err erro
 	return
 }
 
+// Function type for something that calls NewDatabaseContext and wants a callback when the DB is detected
+// to come back online. A rest.ServerContext package cannot be passed since it would introduce a circular dependency
+type DBOnlineCallback func(dbContext *DatabaseContext)
+
 // Creates a new DatabaseContext on a bucket. The bucket will be closed when this context closes.
-func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, options DatabaseContextOptions) (*DatabaseContext, error) {
+func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, options DatabaseContextOptions, online DBOnlineCallback) (*DatabaseContext, error) {
```

Comment:
Great catch!  Yeah this cleans things up considerably, and I think it's a great overall pattern.  Fixed in 308158b09f3c3f38a133e0faf257d8bb7dbf2809